cmake/configure/configure_adolc.cmake

```diff
@@ -0,0 +1,61 @@
+## ---------------------------------------------------------------------
+##
+## Copyright (C) 2016 by the deal.II authors
+##
+## This file is part of the deal.II library.
+##
+## The deal.II library is free software; you can use it, redistribute
+## it, and/or modify it under the terms of the GNU Lesser General
+## Public License as published by the Free Software Foundation; either
+## version 2.1 of the License, or (at your option) any later version.
+## The full text of the license can be found in the file LICENSE at
+## the top level of the deal.II distribution.
+##
+## ---------------------------------------------------------------------
+
+#
+# Configuration for the ADOL-C library:
+#
+
+MACRO(FEATURE_ADOLC_FIND_EXTERNAL var)
+  FIND_PACKAGE(ADOLC)
+
+  IF(ADOLC_FOUND)
+    #
+    # So, we have found a Adol-C library. Let's check whether we can use it.
+    #
+    SET(${var} TRUE)
+
+    #
+    # Look for adolc_settings.h - we'll query it to determine supported features:
+    #
+    DEAL_II_FIND_FILE(ADOLC_SETTINGS_H adolc_settings.h
+      HINTS ${ADOLC_INCLUDE_DIRS} "${ADOLC_INCLUDE_DIRS}/adolc/internal"
+      NO_DEFAULT_PATH NO_CMAKE_ENVIRONMENT_PATH NO_CMAKE_PATH
+      NO_SYSTEM_ENVIRONMENT_PATH NO_CMAKE_SYSTEM_PATH NO_CMAKE_FIND_ROOT_PATH
+      )
+
+    IF(EXISTS ${ADOLC_SETTINGS_H})
+      #
+      # Check whether ADOL-C is configured with extra trig functions
+      #
+      FILE(STRINGS "${ADOLC_SETTINGS_H}" ADOLC_ATRIG_ERF_STRING
+        REGEX "#define ATRIG_ERF")
+      IF(NOT "${ADOLC_ATRIG_ERF_STRING}" STREQUAL "")
+        SET(DEAL_II_ADOLC_WITH_ATRIG_ERF TRUE)
```

Comment:
Can you please move the detection of `DEAL_II_ADOLC_WITH_ATRTG_REF` and `DEAL_II_ADOLC_WITH_ADVANCED_BRANCHING` into `FindADOLC.cmake`? Strip the `DEAL_II_` in the process.

In this file please add
```
SET(DEAL_II_ADOLC_WITH_[...] ${ADOLC_[...]})
SET(DEAL_II_ADOLC_WITH_[...] ${ADOLC_[...]})
```